File: src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
    //用户信息分页展示
    this.findPage=function (pageNum,pageSize) {
        return $http.post('../user/findPage.do?pageNum='+pageNum+"&pageSize="+pageSize);
    }
    this.findUserAll=function(){
        return $http.post('../user/findAll.do');
    }
    //三级联动查询
    this.findCityByPid=function (pid) {
        return  $http.post('../user/findCityByPid.do?pid='+pid);
    }

    //查询实体
    this.findOne=function(id){
        return $http.post('../user/findOne.do?uid='+id);
    }
    //修改和添加
    this.save=function (entity) {
        return  $http.post('../user/save.do',entity);
    }
    //删除
    this.dele=function(ids){
        return $http.post('../user/delete.do',ids);
    }
});

File: src/main/java/com/lening/service/UserService.java
package com.lening.service;

import com.lening.entity.CityBean;
import com.lening.entity.UserBean;
import com.lening.utils.PageResult;

import java.util.List;

public interface UserService {

    boolean register(UserBean userBean);

    PageResult findPageHelper(Integer pageNum, Integer pageSize);

    PageResult findPage(Integer pageNum, Integer pageSize);

    List<CityBean> findCityByPid(Integer pid);

    UserBean findOne(Long uid);

    void save(UserBean userBean);

    void deletes(Long[] ids);

    List<UserBean> findAll();

    PageResult findPageHelperAndCon(UserBean usermohu, Integer pageNum, Integer pageSize);

    UserBean userActivation(String ucode);

    void upaUser(UserBean userBean);

    UserBean findUserByName(String uname);

    UserBean findUserByTel(Long tel);
}
